plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -69,14 +69,26 @@ def set_secrets
       expand_secret_annotations
     end
 
+    # TODO: rename to build_selectors
     def images
-      all = containers.reject { |c| c[:'samson/dockerfile'] == 'none' }
+      all = containers
       modify_init_container { |containers| all += containers }
-      all.map { |c| c.fetch(:image) }.uniq
+      all.map { |c| build_selector_for_container(c) }.compact
     end
 
     private
 
+    def build_selector_for_container(container)
+      if project.docker_image_building_disabled?
+        # TODO: weird that there is no opt-out here
+        [nil, container.fetch(:image)]
+      else
+        selected = container[:"samson/dockerfile"] || 'Dockerfile'
+        next if selected == 'none'
```

Comment:
This is causing an error. Can't use next when not in a loop.

Reply:
just need your time for the review, I can debug travis ;)